sdk-core/src/main/java/com/reach5/identity/sdk/core/utils/Pkce.kt

```diff
@@ -0,0 +1,52 @@
+package com.reach5.identity.sdk.core.utils
+
+import android.os.Parcelable
+import android.util.Base64
+import kotlinx.android.parcel.IgnoredOnParcel
+import kotlinx.android.parcel.Parcelize
+import java.security.MessageDigest
+import java.security.SecureRandom
+import kotlin.text.Charsets.UTF_8
+
+@Parcelize
+class Pkce(val codeVerifier: String) : Parcelable {
+    @IgnoredOnParcel
+    val codeChallenge: String
+    @IgnoredOnParcel
+    val codeChallengeMethod: String
+
+    init {
+        this.codeChallenge = generateCodeChallenge(codeVerifier)
+        this.codeChallengeMethod = "S256"
+    }
+
+    override fun toString(): String {
+        return "code_verifier=$codeVerifier, code_challenge=$codeChallenge, code_challenge_method=$codeChallengeMethod"
+    }
+
+    companion object {
+        fun generate(): Pkce {
```

Comment:
```
fun generate(): Pkce = generateCodeVerifier().let { Pkce(it) }
```

Reply:
I think this syntax is less redundant `Pkce(generateCodeVerifier())`

Reply:
Absolutely 😆